lib/view_component/slotable.rb

```diff
@@ -356,9 +356,14 @@ def set_slot(slot_name, slot_definition = nil, *args, &block)
 
       # If class
       if slot_definition[:renderable]
+        item_options = set_slot_iterator(args, slot_definition[:renderable], slot_name)
+        args.last.merge!(item_options) if item_options.present?
```

Comment:
I think it might be better to avoid mutation here? If I'm understanding the code correctly this would mutate a user's hash that they pass in, which could lead to surprising behavior for the user.

Maybe something like this?

```ruby
args = args.dup.tap{|a| a[-1] = a[-1].merge(item_options) } if item_options.present?
```